catroid/src/main/java/org/catrobat/catroid/utils/ImportProjectsFromExternalStorage.java

```diff
@@ -0,0 +1,235 @@
+				if (fileExists) {
+					String newFileName = uniqueNameProvider.getUniqueName(srcFileName, scope);
+					copyProjectsFromExternalToInternalStorage(srcFile, new File(dstDir, newFileName));
+				} else {
+					copyProjectsFromExternalToInternalStorage(srcFile, new File(dstDir, srcFileName));
+				}
+			} else {
+				transferData(srcFile, new File(dstDir, srcFileName));
+			}
+		}
+
+		return dstDir;
+	}
+
+	private static boolean copyProjects() {
+		// delete tmp folders before start copying
+		File tmpDir = new File(OLD_EXTERNAL_STORAGE_ROOT_DIRECTORY, "/tmp");
+
+		if (tmpDir.exists()) {
+			try {
+				StorageOperations.deleteDir(tmpDir);
+			} catch (IOException e) {
+				Log.e(TAG, e.getMessage());
+			}
+		}
+		// copy projects to internal storage
+		try {
+			copyProjectsFromExternalToInternalStorage(OLD_EXTERNAL_STORAGE_ROOT_DIRECTORY, FlavoredConstants.DEFAULT_ROOT_DIRECTORY);
+			return true;
+		} catch (IOException e) {
+			Log.e(TAG, e.getMessage());
+			return false;
+		}
+	}
+
+	private static boolean deleteOldExternalPocketCodeFolder() {
+		try {
+			StorageOperations.deleteDir(OLD_EXTERNAL_STORAGE_ROOT_DIRECTORY);
+			return true;
+		} catch (IOException e) {
+			Log.e(TAG, e.getMessage());
+			return false;
+		}
+	}
+
+	public void importOldPocketCodePrograms(Context context) {
+		if (isAppInInternalStorageInstalled(context) && OLD_EXTERNAL_STORAGE_ROOT_DIRECTORY.exists()) {
+			showImportProjectsDialog(activity);
+		}
+	}
+
+	private AlertDialog.Builder createImportProjectsDialog(Activity activity) {
+		AlertDialog.Builder alertDialog = new AlertDialog.Builder(activity);
+		alertDialog.setTitle(R.string.import_dialog_title)
+				.setCancelable(false)
+				.setMessage(R.string.import_dialog_message)
+				.setPositiveButton(R.string.import_dialog_move_btn, this)
+				.setNeutralButton(R.string.import_dialog_copy_btn, this);
+		return alertDialog;
+	}
+
+	private void showImportProjectsDialog(Activity activity) {
+		AlertDialog.Builder alertDialog = createImportProjectsDialog(activity);
+		boolean showDialog = PreferenceManager
+				.getDefaultSharedPreferences(activity).getBoolean(SHOW_COPY_PROJECTS_FROM_EXTERNAL_STORAGE_DIALOG, true);
+		if (showDialog) {
+			alertDialog.show();
+		}
+	}
+
+	@Override
+	public void onClick(DialogInterface dialog, int which) {
+		switch (which) {
+			case DialogInterface.BUTTON_POSITIVE:
+				CopyProjectsTask moveProjectsTask = new CopyProjectsTask(activity, true);
+				moveProjectsTask.execute();
+				doNotShowDialogAgain();
+				break;
+			case DialogInterface.BUTTON_NEUTRAL:
+				CopyProjectsTask copyProjectsTask = new CopyProjectsTask(activity, false);
+				copyProjectsTask.execute();
+				doNotShowDialogAgain();
+				break;
+		}
+	}
+
+	private void doNotShowDialogAgain() {
+		PreferenceManager.getDefaultSharedPreferences(activity).edit().putBoolean(SHOW_COPY_PROJECTS_FROM_EXTERNAL_STORAGE_DIALOG, false).apply();
+	}
+
+	private boolean isAppInInternalStorageInstalled(Context context) {
```

Comment:
why is this necessary? since we do not offer the internal storage as an option, everyone having this released code will be saving projects in internal storage.

Reply:
deleted